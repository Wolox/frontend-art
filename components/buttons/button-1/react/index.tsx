import React from 'react';

import '../styles.scss';

function Button1() {
  return (
    <div className="column center btn-group-1">
      <button className="btn btn-1 m-bottom-2" disabled={true}>
        <div className="btn__content">cancel</div>
      </button>
      <button type="button" className="btn btn-1">
        <div className="btn__content">submit</div>
      </button>
    </div>
  );
}

export default Button1;
