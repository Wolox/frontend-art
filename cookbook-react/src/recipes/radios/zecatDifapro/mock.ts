export const staticProps = {
  extraInfo: '$ 50',
  id: 'test',
  info: 'Peque√±a',
  name: 'test',
  title: 'Hamburguesa'
};
