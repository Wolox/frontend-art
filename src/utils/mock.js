export default {
  title: 'Checkbox 2',
  html: `
  <div class="item-checkbox">
    <div class="checkbox-2">
      <input id="checkbox-2" class="checkbox" type="checkbox" />
      <label for="checkbox-2">
        <svg width="18px" height="18px" viewbox="0 0 18 18">
          <path d="M1,9 L1,3.5 C1,2 2,1 3.5,1 L14.5,1 C16,1 17,2 17,3.5 L17,14.5 C17,16 16,17 14.5,17 L3.5,17 C2,17 1,16 1,14.5 L1,9 Z" />
          <polyline points="1 9 7 14 15 4" />
        </svg>
      </label>
    </div>
  </div>
  `,
  css: `
  .item-checkbox {
    --primary-color: #636060;
    --secondary-color: #FFF;
    --active-color: #00D478;
  }

  .checkbox {
    display: none;
  }  
  
  .checkbox-2 > label {
    cursor: pointer;
    height: 18px;
    margin: 0 auto;
    position: relative;
    transform: translate3d(0, 0, 0);
    width: 18px;
 }
  .checkbox-2 > label::before {
    background-color: rgba(255, 255, 255, 0.2);
    border-radius: 50%;
    content: '';
    height: 48px;
    left: -15px;
    opacity: 0;
    position: absolute;
    top: -15px;
    transition: opacity 330ms cubic-bezier(0.79, 0.01, 0.375, 0.995);
    width: 48px;
 }
  .checkbox-2 > label svg {
    fill: none;
    position: relative;
    stroke: var(--primary-color);
    stroke-linecap: round;
    stroke-linejoin: round;
    stroke-width: 1.5;
    transform: translate3d(0, 0, 0);
    transition: all 330ms cubic-bezier(0.79, 0.01, 0.375, 0.995);
    z-index: 1;
 }
  .checkbox-2 > label svg path {
    stroke-dasharray: 60;
    stroke-dashoffset: 0;
 }
  .checkbox-2 > label svg polyline {
    stroke-dasharray: 22;
    stroke-dashoffset: 66;
 }
  .checkbox-2 > label:hover::before {
    opacity: 1;
 }
  .checkbox-2 > label:hover svg {
    stroke: var(--active-color);
 }
  .checkbox-2 > input:checked + label svg {
    stroke: var(--active-color);
 }
  .checkbox-2 > input:checked + label svg path {
    stroke-dashoffset: 60;
    transition: all 0.3s linear;
 }
  .checkbox-2 > input:checked + label svg polyline {
    stroke-dashoffset: 42;
    transition: all 0.2s linear;
    transition-delay: 0.15s;
 }
 
  `,
  author: 'FrankieCapo123'
}