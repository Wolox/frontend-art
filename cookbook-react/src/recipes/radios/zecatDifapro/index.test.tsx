import React from 'react';
import { render, screen } from '@testing-library/react';

import { staticProps } from './mock';

import ZecatDifapro from '.';

jest.mock('i18next', () => ({
  t: (key: string) => key
}));

describe('ZecatDifapro test', () => {
  test('Should match unchecked radio snapshot', () => {
    const { container } = render(<ZecatDifapro {...staticProps} />);
    expect(container).toMatchSnapshot();
  });

  test('Should render unchecked radio.', () => {
    render(<ZecatDifapro {...staticProps} />);
    const input = screen.getByTestId(`${staticProps.name}-zecatdifapro`);
    expect(input).toHaveClass('customRadio');
    expect(input).not.toHaveClass('checked');
  });

  test('Should render checked radio.', () => {
    render(<ZecatDifapro {...staticProps} checked />);
    const input = screen.getByTestId(`${staticProps.name}-zecatdifapro`);
    expect(input).toHaveClass('customRadio');
    expect(input).toHaveClass('checked');
  });

  test('Should render unchecked checkbox.', () => {
    render(<ZecatDifapro {...staticProps} type="checkbox" />);
    const input = screen.getByTestId(`${staticProps.name}-zecatdifapro`);
    expect(input).toHaveClass('customCheckbox');
    expect(input).not.toHaveClass('checked');
  });

  test('Should render checked checkbox.', () => {
    render(<ZecatDifapro {...staticProps} checked type="checkbox" />);
    const input = screen.getByTestId(`${staticProps.name}-zecatdifapro`);
    expect(input).toHaveClass('customCheckbox');
    expect(input).toHaveClass('checked');
  });

  test('Should render custom image for unchecked radio', () => {
    render(<ZecatDifapro {...staticProps} inputStyle="custom" />);
    expect(screen.getByAltText('ZecatDifapro:notSelectedImg')).toBeInTheDocument();
  });

  test('Should render custom image for checked radio', () => {
    render(<ZecatDifapro {...staticProps} inputStyle="custom" checked />);
    expect(screen.getByAltText('ZecatDifapro:selectedImg')).toBeInTheDocument();
  });

  test('Should render custom image for unchecked checkbox', () => {
    render(<ZecatDifapro {...staticProps} type="checkbox" inputStyle="custom" />);
    expect(screen.getByAltText('ZecatDifapro:notSelectedImg')).toBeInTheDocument();
  });

  test('Should render custom image for checked checkbox', () => {
    render(<ZecatDifapro {...staticProps} type="checkbox" inputStyle="custom" checked />);
    expect(screen.getByAltText('ZecatDifapro:selectedImg')).toBeInTheDocument();
  });
});
