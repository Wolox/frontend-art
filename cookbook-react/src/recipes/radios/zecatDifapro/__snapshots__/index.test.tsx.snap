// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZecatDifapro test Should match unchecked radio snapshot 1`] = `
<div>
  <div
    class="wrapper default row middle"
  >
    <div
      class="firstSection row"
    >
      <input
        class="customRadio"
        data-testid="test-zecatdifapro"
        id="test"
        name="test"
      />
      <div
        class="info column"
      >
        <label
          class="title"
          for="test"
        >
          Hamburguesa
        </label>
        <label
          for="test"
        >
          Peque√±a
        </label>
      </div>
    </div>
    <label
      class="extraInfo"
      for="test"
    >
      $ 50
    </label>
  </div>
</div>
`;
